File: apps/mobile/lib/domain/models/itinerary.dart
class ItineraryModel {
  final String id;
  final String tripId;
  final String destinationName;
  final DateTime? plannedArrival;
  final DateTime? plannedDeparture;
  final double latitude;
  final double longitude;
  final int sequenceOrder;

  ItineraryModel({
    required this.id,
    required this.tripId,
    required this.destinationName,
    this.plannedArrival,
    this.plannedDeparture,
    required this.latitude,
    required this.longitude,
    required this.sequenceOrder,
  });

  factory ItineraryModel.fromJson(Map<String, dynamic> json) {
    return ItineraryModel(
      id: json['id'] as String,
      tripId: json['trip_id'] as String,
      destinationName: json['destination_name'] as String,
      plannedArrival: json['planned_arrival'] != null ? DateTime.parse(json['planned_arrival']) : null,
      plannedDeparture: json['planned_departure'] != null ? DateTime.parse(json['planned_departure']) : null,
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      sequenceOrder: json['sequence_order'] as int? ?? 1,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'destination_name': destinationName,
      'latitude': latitude,
      'longitude': longitude,
      'sequence_order': sequenceOrder,
    };
  }
}

File: backend/app/domain/models/itinerary.py
import uuid
from datetime import datetime
from typing import TYPE_CHECKING, Optional
from sqlalchemy import String, Integer, Float, DateTime, ForeignKey, Uuid
from sqlalchemy.orm import Mapped, mapped_column, relationship
from backend.app.domain.models.base import UUIDModel

if TYPE_CHECKING:
    from backend.app.domain.models.trip import Trip


class Itinerary(UUIDModel):
    __tablename__ = "itineraries"

    trip_id: Mapped[uuid.UUID] = mapped_column(
        Uuid(as_uuid=True),
        ForeignKey("trips.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    destination_name: Mapped[str] = mapped_column(
        String(255),
        nullable=False,
    )
    planned_arrival: Mapped[Optional[datetime]] = mapped_column(
        DateTime(timezone=True),
        nullable=True,
    )
    planned_departure: Mapped[Optional[datetime]] = mapped_column(
        DateTime(timezone=True),
        nullable=True,
    )
    latitude: Mapped[float] = mapped_column(
        Float,
        nullable=False,
    )
    longitude: Mapped[float] = mapped_column(
        Float,
        nullable=False,
    )
    sequence_order: Mapped[int] = mapped_column(
        Integer,
        default=1,
        nullable=False,
    )

    # Relationships
    trip: Mapped["Trip"] = relationship(
        "Trip",
        back_populates="itineraries",
    )
